hyperspy/learn/mva.py

```diff
@@ -1013,13 +1015,30 @@ def plot_explained_variance_ratio(self, n=None, log=True, threshold=0,
         # Determine right number of components for signal and cutoff value
         if isinstance(threshold, float):
```

Comment:
Ok, should be fixed. I also fixed an edge-case bug I found if the `threshold` value was too small (less than the minimum variance)